datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx

```diff
@@ -0,0 +1,327 @@
+        const onPreventMouseDown = (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+        };
+        return (
+            <CustomTag
+                onMouseDown={onPreventMouseDown}
+                closable={closable}
+                onClose={onClose}
+                style={{
+                    padding: selectedType === EntityType.Tag ? '0px 7px 0px 0px' : '0px 7px',
+                    marginRight: 3,
+                    borderRadius: '100em',
+                    lineHeight: 0,
+                    display: 'flex',
+                    justifyContent: 'start',
+                    alignItems: 'center',
+                }}
+            >
+                {label}
+            </CustomTag>
+        );
+    };
+
+    // Handle the Enter press
+    useEnterKeyListener({
+        querySelectorToExecuteClick: '#addTagButton',
+    });
+
+    if (showCreateModal) {
+        return (
+            <CreateTagModal
+                visible={visible}
+                onClose={onCloseModal}
+                onBack={() => setShowCreateModal(false)}
+                tagName={inputValue}
+                entityUrn={entityUrn}
+                entitySubresource={entitySubresource}
+            />
+        );
+    }
+
+    const getUrnId = (urnId: string) => {
+        const { name: selectedName, type: selectedType } = getSelectedValue(urnId);
+        let tempUrn = '';
+        if (selectedType === EntityType.Tag) {
+            tempUrn = `urn:li:tag:${selectedName}`;
+        }
+        if (selectedType === EntityType.GlossaryTerm) {
+            tempUrn = `urn:li:glossaryTerm:${selectedName}`;
+        }
+        return tempUrn;
+    };
+
+    // When a Tag or term search result is selected, add the urn to the UrnIds
+    const onSelectValue = (newUrnId: string) => {
+        if (newUrnId === CREATE_TAG_VALUE) {
+            setShowCreateModal(true);
+            return;
+        }
+        const newUrnIds = [...(urnIds || []), getUrnId(newUrnId)];
+        setUrnIds(newUrnIds);
+    };
+
+    // When a Tag or term search result is deselected, remove the urn from the Owners
+    const onDeselectValue = (urnId: string) => {
+        const newUrnIds = urnIds?.filter((u) => u !== getUrnId(urnId));
+        setUrnIds(newUrnIds);
+    };
+
+    // Function to handle the modal action's
+    const onOk = () => {
+        let mutation: ((input: any) => Promise<any>) | null = null;
+        if (type === EntityType.Tag) {
+            mutation = addTagsMutation;
+        }
+        if (type === EntityType.GlossaryTerm) {
+            mutation = addTermsMutation;
+        }
+
+        if (!entityUrn || !mutation) {
+            onCloseModal();
+            return;
+        }
+        setDisableAdd(true);
+
+        let input = {};
+        let actionType = EntityActionType.UpdateSchemaTags;
+        if (type === EntityType.Tag) {
+            input = {
+                tagUrns: urnIds,
```

Comment:
Again, let's never use the name "urnIds" --> URNs are already implying that they are ids.

Just call them "urns"

Reply:
Changed and committed